palace/utils/geodata.cpp

```diff
@@ -762,52 +763,46 @@ BoundingBox BoundingBoxFromPointCloud(MPI_Comm comm,
     // two opposite edges of the cuboid. Now look for a component that maximizes distance
     // from the planar system: complete the axes with a cross, then use a dot product to
     // pick the greatest deviation.
-    const Eigen::Vector3d n_3 = n_1.cross(n_2).normalized();
-
-    auto OutOfPlaneComp = [&n_1, &n_2, &n_3, &origin](const auto &v_1, const auto &v_2)
+    auto OutOfPlaneDistance = [&n_1, &n_2, &origin](const Eigen::Vector3d &v)
     {
-      // Precedence of directions is in reverse order of discovery of the directions. The
-      // most important deciding feature is the distance in the out of plane direction,
-      // then in the first off-diagonal, then finally in the diagonal direction.
-      const std::array<double, 3> dist_1{(v_1 - origin).dot(n_3), (v_1 - origin).dot(n_2),
-                                         (v_1 - origin).dot(n_1)};
-      const std::array<double, 3> dist_2{(v_2 - origin).dot(n_3), (v_2 - origin).dot(n_2),
-                                         (v_2 - origin).dot(n_1)};
-      return dist_1 < dist_2;
+      return ((v - origin) - (v - origin).dot(n_1) * n_1 - (v - origin).dot(n_2) * n_2)
+          .norm();
     };
 
-    // There is a degeneration if the final point is within the plane defined by (v_000,
-    // v_001, v_111).
-    const auto &t_1 = *std::max_element(vertices.begin(), vertices.end(), OutOfPlaneComp);
-    constexpr double planar_tolerance = 1.0e-9;
-    box.planar = std::abs(n_3.dot(t_0)) < planar_tolerance &&
-                 std::abs(n_3.dot(t_1)) < planar_tolerance;
-    if (!box.planar)
-    {
-      MFEM_VERIFY(&t_0 != &t_1, "Degenerate coordinates!");
-      MFEM_VERIFY(&t_0 != &v_000, "Degenerate coordinates!");
-      MFEM_VERIFY(&t_0 != &v_111, "Degenerate coordinates!");
-      MFEM_VERIFY(&t_1 != &v_000, "Degenerate coordinates!");
-      MFEM_VERIFY(&t_1 != &v_111, "Degenerate coordinates!");
-    }
-
-    // If t_1 points to v_000, t_0 or v_111, then the data is coplanar. Establish if t_0 is
-    // a diagonal or not (using Pythagoras). Only pick t_1 for v_001 if the points are non-
-    // planar, and t_0 is longer.
-    bool t_0_gt_t_1 = t_0.norm() >= t_1.norm();
-    const auto &v_001 = !box.planar && t_0_gt_t_1 ? t_1 : t_0;
-    const auto &v_011 = !box.planar && t_0_gt_t_1 ? t_0 : t_1;
+    // Collect the furthest point from the plane.
+    auto max_distance = OutOfPlaneDistance(
+        *std::max_element(vertices.begin(), vertices.end(),
+                          [OutOfPlaneDistance](const auto &x, const auto &y)
+                          { return OutOfPlaneDistance(x) < OutOfPlaneDistance(y); }));
+
+    constexpr double rel_tol = 1e-3;
```

Comment:
Again, this seems like a very large relative tolerance. The previous (absolute) tolerance was 1e-9 which I understand caused some issues, but does it make sense to have relative tolerances be 1e-6 or so?